src/dash-table/components/CellFactory.tsx

```diff
@@ -163,12 +214,10 @@ export default class CellFactory {
         wrappers,
         styles,
         contents,
-        (w, s, c, i, j) => React.cloneElement(w, {
-            children: [c],
-            style: R.mergeAll([
-                s,
-                edges && edges.getStyle(i, j)
-            ])
-        })
+        (w, s, c, i, j) => {
+            const edge = edges && edges.getStyle(i, j);
+
+            return this.getDataCell.get(i, j)(w, c, s, edge && edge.borderBottom, edge && edge.borderLeft, edge && edge.borderRight, edge && edge.borderTop);
```

Comment:
Same as for op cells but for the data cells themselves.

The same has not been done for headers and filters as, while it will improve performance slightly, the bulk of the gains are in the data.